include/dirac_quda.h

```diff
@@ -501,6 +503,8 @@ namespace quda {
   class DiracWilson : public Dirac {
 
   protected:
+    double distance_pc_alpha0;
```

Comment:
Should these parameters also be added to `DiracParam` so that they can set when a Dirac operator is constructed?

Reply:
I added them to `DiracParam` in my early version of the implementation. But adding them to `DiracParam` means we will always get a Dirac operator with distance preconditioning if corresponding parameters are set in `QudaInvertParam`. This lead to some issues in functions like `MatQuda` and `DslashQuda`, which will apply a preconditioned M or Dslash to the input.
Of course, we could always apply the corresponding reweight to input and output fields. I thought it was not a clean implementation then, so I remove these parameters in `DiracParam`.
Back to now, I think setting parameters by an extra interface `setDistancePrecondition` is also not clean. I can add these things back to `DiracParam`.